src/js/view/controls/components/menu/settings-menu.js

```diff
@@ -52,7 +42,7 @@ const SettingsMenu = (api, model, controlbar, localization) => {
     // Quality Tracks
     model.change('levels', (changedModel, levels) => {
         setLevelsMenu(levels);
-    }, settingsMenu);
+    });
```

Comment:
It seems like all of these model listeners are leaky. There is nothing removing these listeners when the menu is removed and destroyed - is there? So the references to the menus live on in the model listeners and when new menus are created new listeners are added.

The way to test this is set up a player with a playlist that has items with different menus (one playlist item has levels, one does not.) Then go through the playlist a few times to see if this model's listeners increase in number over time or level out.

Reply:
Usually we would pass the model into the menu instance, and have it register listeners with a context of `this`. Then in `menu.destroy` all listeners bound to the instance are removed:

```
class ViewItem {

  constructor(model) {
     this.model = model;
     model.on('change:x', this.onChangeX, this);
     model.on('change:y', this.onChangeY, this);
  }

  destroy () {
    if (this.model) {
       this.model.off(null, null, this);
   }
}
```

Reply:
Couldn't agree more, here's my thoughts on that: https://github.com/jwplayer/jwplayer/pull/3565